website/pages/docs/builders/vmware/vsphere-clone.mdx

```diff
@@ -42,6 +42,43 @@ necessary for this build to succeed and can be found further down the page.
 
 @include 'builder/vsphere/clone/CloneConfig-not-required.mdx'
 
+### vApp Options Configuration
+
+@include 'builder/vsphere/clone/vAppConfig-not-required.mdx'
+
+Example of usage:
+
+<Tabs>
+<Tab heading="JSON">
+
+```json
+ "vapp": {
+     "properties": {
+         "hostname": "{{ user `hostname`}}",
+         "user-data": "{{ env `USERDATA`}}"
+     }
+ }
+```
+
+A `user-data` field requires the content of a yaml file to be encoded with base64. This
+can be done via environment variable:
+`export USERDATA=$(gzip -c9 <userdata.yaml | { base64 -w0 2>/dev/null || base64; })`
+
+</Tab>
+<Tab heading="HCL2">
+
+```hcl
+   vapp {
+     properties = {
+        hostname  = var.hostname
+        user-data = base64encode(var.user_data)
+     }
```

Comment:
Yay HCL2 makes that so much simpler 👍

Reply:
yes!!! I loved it! That's why I wanted to use this example 😆

Reply:
😆 also now that I think of it, we could totally have a base64encode func in Packer JSON 🤔